packages/api-derive/src/chain/util.ts

```diff
@@ -25,35 +25,41 @@ export function createBlockNumberDerive <T extends { number: Compact<BlockNumber
     );
 }
 
-export function getAuthorDetails (header: Header, queryAt: QueryableStorage<'rxjs'>): Observable<[Header, Vec<AccountId> | null, AccountId | null]> {
-  // this is Moonbeam specific
-  if (queryAt.authorMapping && queryAt.authorMapping.mappingWithDeposit) {
-    const mapId = header.digest.logs[0] && (
-      (header.digest.logs[0].isConsensus && header.digest.logs[0].asConsensus[1]) ||
-      (header.digest.logs[0].isPreRuntime && header.digest.logs[0].asPreRuntime[1])
-    );
-
-    if (mapId) {
-      return combineLatest([
-        of(header),
-        queryAt.session
-          ? queryAt.session.validators()
-          : of(null),
-        queryAt.authorMapping.mappingWithDeposit<IOption<{ account: AccountId } & Codec>>(mapId).pipe(
-          map((opt) =>
-            opt.unwrapOr({ account: null }).account
-          )
+export function getAuthorDetails(header: Header, queryAt: QueryableStorage<'rxjs'>): Observable<[Header, Vec<AccountId> | null, AccountId | null]> {
+  // Some consensus systems store author info in a digest item, e.g. Nimbus Consensus and can directly return the block author
+  let author_accountId: Observable<AccountId | null> = of(null);
+  // the consensusId from the digest must be mapped to a corresponding accountId first. The following is Nimbus specific
+  const author_consensusId = header.digest.logs[0] && (
```

Comment:
```suggestion
  const authorConsensusId = header.digest.logs[0] && (
```

camelCase (as above)